eventuals/task.h

```diff
@@ -465,15 +498,37 @@ class TaskWith {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename Value_>
-class Task : public TaskWith<Value_> {
+template <typename... Types_>
+class Task : public TaskWith<Types_...> {
+ public:
+  template <typename... Args_>
+  using With = TaskWith<Types_..., Args_...>;
+
+  template <typename F>
+  Task(F f)
+    : TaskWith<Types_...>(std::move(f)) {}
+};
```

Comment:
```suggestion
template <typename...>
class Task;
```

Let's not allow for an arbitrary number of types but restrict people to use the partial instantiations below.